lib/eex/lib/eex.ex

```diff
@@ -103,7 +103,7 @@ defmodule EEx do
 
   @type line :: non_neg_integer
   @type column :: non_neg_integer
-  @type marker :: '=' | '/' | '|' | ''
+  @type marker :: [?=] | [?/] | [?|] | []
```

Comment:
```suggestion
  @type marker :: [?=] | [?/] | [?|] | []
```

Wouldn't this work as `~c"="`?

Reply:
I thought so too, I didn't dive into why it fails, but I'm getting this error:

```
== Compilation error in file lib/eex.ex ==
** (CompileError) lib/eex.ex:106: invalid binary specification, expected <<_::size>>, <<_::_*unit>>, or <<_::size, _::_*unit>> with size being non-negative integers, and unit being an integer between 1 and 256
    (elixir 1.15.0-dev) lib/kernel/typespec.ex:936: Kernel.Typespec.compile_error/2
    (stdlib 4.0) lists.erl:1462: :lists.mapfoldl_1/3
    (elixir 1.15.0-dev) lib/kernel/typespec.ex:845: Kernel.Typespec.typespec/4
    (stdlib 4.0) lists.erl:1462: :lists.mapfoldl_1/3
    (elixir 1.15.0-dev) lib/kernel/typespec.ex:464: Kernel.Typespec.typespec/4
    (elixir 1.15.0-dev) lib/kernel/typespec.ex:307: Kernel.Typespec.translate_type/2
    (stdlib 4.0) lists.erl:1462: :lists.mapfoldl_1/3
    (stdlib 4.0) lists.erl:1463: :lists.mapfoldl_1/3
```

Maybe we are missing an expansion step when translating typespecs?

Reply:
I will take a look.

Reply:
Actually, it is good it doesn't work, because `'foo' | 'bar'` is not `foo` or `bar`, but any charlist with any amount of `?f` and `?o` or any charlist with any amount of `?b`, `?a`, and `?r`. In our case, we are also allowing `'=='`, which is not correct, but it is the best we can do right now. So this is better as is of now. :)